### ISSUE TITLE
Export Song Play History does not work when a session is specified.

### ISSUE THREAD

**Issue by sharmanr** [mpogue2/SquareDesk#724]
There are 2 problems:

1. The table column is specified as session_id but it should be session_rowid.  This results in an SQL “Parameter error” and so no data is returned (the resultant file just has a header line).

2. When that is fixed there is a problem if session are changed from the defaults.  This is because when the checkbox is examined the session_id is set to the index in the list, which is not necessarily the actual session_id.   For example, delete all but the first session then create a new one.  Select this and you should get index 7 but you get index 2, so again no songs match.

See songhistory.patch for fixes for both.
[songhistory.patch.patch.txt](https://github.com/mpogue2/SquareDesk/files/9935775/songhistory.patch.patch.txt)

**Comment by mpogue2** [comment]
Should be fixed now, as per your patch.  I tested as far as I could, given that I don't use this feature....
Please test, and let me know if it works correctly, as per your testing!

**Comment by danlyke** [comment]
Sorry about dropping the ball on that code review, still trying to find the
time to figure out whatever the probable race condition is on Linux.

**Comment by sharmanr** [comment]
Hi Mike.

Yes, it works fine.   (I tested with two “new” session id’s that would have failed before,  and with both dates and no dates,  then checked with “all session” and all is well.

I do have a question or two about building it if you don’t mind…

I downloaded the source,  did the make of soundtouch as per README, but navigated to Finder the “test123.pro” file right clicked on it and selected “open with Qt Creator”.  It wanted to configure the project, which I did,  but when I clicked on build I got this error:

:-1: error: No rule to make target `/Users/richard/src/SquareDesk-master/SquareDesk-DEV/build-test123-Desktop_arm_darwin_generic_mach_o_64bit-Debug/../sdlib/libsdlib.a', needed by `SquareDesk.app/Contents/MacOS/SquareDesk'.  Stop.

So I exited,  manually opened Qt Creator, and used “open project” and the build worked.   Why the different behaviour?

The reason I did it that way was to attempt to avoid it asking if Qt Creator can access Downloads and Documents.  Whether I say Allow or Deny it asks me each time, even though in System Preferences / Privacy Qt Creator is checked under both “Full Disk Access” and “Files and Folders”.  Is there any way to stop it asking each time?

Cheers,
Richard
